packages/next/src/client/components/app-router.tsx

```diff
@@ -512,15 +515,25 @@ function Router({
       </HotReloader>
     )
   } else {
-    // In production, we only apply the user-customized global error boundary.
-    content = (
-      <ErrorBoundary
-        errorComponent={globalError[0]}
-        errorStyles={globalError[1]}
-      >
-        {content}
-      </ErrorBoundary>
-    )
+    // If gracefully degrading is applied in production,
+    // wrap app with the graceful error boundary, so it can catch errors rather than GlobalError
+    if (gracefullyDegrade && process.env.NODE_ENV === 'production') {
+      content = (
+        <GracefullyDegradingErrorBoundary>
```

Comment:
I was imagining that in the graceful degradation case, we didn't need to ship any error boundary at all -- just let the error hit the console and skip the boundary all together. Is there a reason to still use `GracefullyDegradingErrorBoundary`?